samples/components/dialects/Hive12Dialect.tdd

```diff
@@ -1176,7 +1217,8 @@
       <argument type='datetime' />
     </date-function>
     <date-function name='DATENAME' return-type='str'>
-      <formula part='week'>FORMAT_NUMBER(FLOOR((14 + DATEDIFF(%2, TRUNC(%2,&apos;YY&apos;)) + DATEDIFF(TRUNC(%2,&apos;YY&apos;),NEXT_DAY(TRUNC(%2,&apos;YY&apos;),&apos;SU&apos;)))/7),0)</formula>
+      <formula part='weekday'>CAST((8 + DATEDIFF(CAST(%2 AS DATE),NEXT_DAY(%2,%3))) AS STRING)</formula>
```

Comment:
`weekday` shouldn't matter for this and the change in `DATEPART` either correct?

Reply:
I think it matters since weekday's formula is different than the regular one, and it is also used the last argument(custom start of week)